### PR TITLE
Fix infix wrap

### LINKED ISSUE #673: Regression: failure to stabilize
```
  let rex =
    Pcre.regexp
      ("^[0-9]{2}"
       (* xxxxxxxxxxx               *)
       ^ "(.{12})"
       (* xxxxxxxxxxxxxxxxxx        *)
       ^ "(.{4})"
       (* xxxxxxxxxxxx              *)
       ^ "([0-9]{3})"
       (* xxxxxxxx                  *)
       ^ "(.{60})"
       (* xxxxxxxxxxxxxxxxxxxx      *)
       ^ "(.{12})"
       (* xxxxxxxxxxxxxxx           *)
       ^ "(.{12})"
       (* xxxxxxxxxxxxxxxxxxx       *)
       ^ "([0-9]{3})"
       (* xxxxxxxxxxxxxxxxxxxxxxxxx *)
       ^ "([0-9]{3})"
       (* xxxxxxxxxxx               *)
       ^ "(.{15})"
       (* xxxxxxxxxxxxxxxxxx        *)
       ^ "([0-9]{7})"
       (* xxxxxxxxxxxxx             *)
       ^ "(.{10})"
       (* xxxxxxxxxxxxx             *)
       ^ date_fmt
       (* xxxxxxxxxxxxx             *)
       ^ "([0-9]{18})"
       (* xxxxx                     *)
       ^ "(.)"
       (* xxxxxxxxxxx               *)
       ^ "([0-9]{3})"
       (* xxxxxxxxxxxxxx            *)
       ^ "(.{15})"
       (* xxxxxxxxxxxxxxxxxxxx      *)
       ^ "(.{3})"
       (* xxxxxxxxxx                *)
       ^ "(.{27})$")
  ;;
```

### PR DESCRIPTION
**Pull request by gpetiot**
Fix #673

### CODE REVIEW

**Comment by gpetiot** [comment]
I would like to have @jberdine or @hhugo 's opinion before merging this one, not sure it's good adding dependencies between options like that.

**Comment by jberdine** [comment]
Agreed, we should avoid such dependencies. Also, I see a large diff with `test_branch`?

**Comment by gpetiot** [comment]
Yes restricting the wrapping is a cheap fix, but the only way I found so far.

By the way 20 iterations are required to stabilize and it is formatted as:
```ocaml
let rex =
  Pcre.regexp
    ( "^[0-9]{2}" (* xxxxxxxxxxx               *) ^ "(.{12})"
    (* xxxxxxxxxxxxxxxxxx        *) ^ "(.{4})"
    (* xxxxxxxxxxxx              *) ^ "([0-9]{3})"
    (* xxxxxxxx                  *) ^ "(.{60})"
    (* xxxxxxxxxxxxxxxxxxxx      *) ^ "(.{12})"
    (* xxxxxxxxxxxxxxx           *) ^ "(.{12})"
    (* xxxxxxxxxxxxxxxxxxx       *) ^ "([0-9]{3})"
    (* xxxxxxxxxxxxxxxxxxxxxxxxx *) ^ "([0-9]{3})"
    (* xxxxxxxxxxx               *) ^ "(.{15})"
    (* xxxxxxxxxxxxxxxxxx        *) ^ "([0-9]{7})"
    (* xxxxxxxxxxxxx             *) ^ "(.{10})"
    (* xxxxxxxxxxxxx             *) ^ date_fmt
    (* xxxxxxxxxxxxx             *) ^ "([0-9]{18})"
    (* xxxxx                     *) ^ "(.)" (* xxxxxxxxxxx               *)
    ^ "([0-9]{3})" (* xxxxxxxxxxxxxx            *) ^ "(.{15})"
    (* xxxxxxxxxxxxxxxxxxxx      *) ^ "(.{3})"
    (* xxxxxxxxxx                *) ^ "(.{27})$" )
```

**Comment by jberdine** [comment]
Wow, 20. Ugh. This is a tough case. The original code really looks like wrapping infix args is not what would be intended, and no matter what the comment attachment does seems like it will generate code that isn't great.

**Comment by gpetiot** [comment]
I changed the wrapping only in presence of comments, the diff of test_branch is much smaller now:
```ocaml
diff --git a/asmcomp/closure_offsets.ml b/asmcomp/closure_offsets.ml
index f371c1e7d..bfbdcf03f 100644
--- a/asmcomp/closure_offsets.ml
+++ b/asmcomp/closure_offsets.ml
@@ -30,7 +30,8 @@ let add_closure_offsets {function_offsets; free_variable_offsets}
     let env_pos =
       let arity = Flambda_utils.function_arity function_decl in
       env_pos + 1 (* GC header; either [Closure_tag] or [Infix_tag] *) + 1
-      (* full application code pointer *) + 1 (* arity *)
+      (* full application code pointer *)
+      + 1 (* arity *)
       + if arity > 1 then 1 else 0
       (* partial application code pointer *)
     in
```

File: src/Fmt_ast.ml [hunk 1]
```diff
@@ -1223,12 +1223,15 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
       match c.conf.break_infix with
       | `Wrap ->
           List.group op_args
-            ~break:(fun (_, _, (_, args1)) (_, _, (_, args2)) ->
+            ~break:(fun (has_cmts, _, _, (_, args1))
+                   (_, _, _, (_, args2))
+                   ->
```
Comment (jberdine) [inline src/Fmt_ast.ml]: Note to us: this formatting is pretty odd.

**Pull request merged**